app/utils/feedback_gen.py

```diff
@@ -1,7 +1,33 @@
 from .llm import llm, sm, um
 
 
-def generate_feedback(user_history, dsl_desc, errors, **kwargs):
+def generate_diff(old_nlr, new_nlr, **kwargs):
+    template_str = f"""You are a developer bot that is developing a program as per the following user instruction. 
+The program plan has been updated.
+Here is the older plan: {old_nlr}
```

Comment:
Why are you not giving user instruction?

Reply:
modified the user message prompt. We are giving both system and user message